src/MarginTrading.SettingsService/Controllers/ScheduleSettingsController.cs

```diff
@@ -153,6 +157,35 @@ public async Task<List<CompiledScheduleContract>> StateList([FromBody] string[]
             return result;
         }
 
+        /// <summary>
+        /// Get current trading status of markets. Platform schedule (with PlatformScheduleMarketId) overrides all others.
+        /// </summary>
+        /// <param name="marketIds">Optional. List of market Id's.</param>
+        [HttpPost]
+        [Route("markets-status")]
+        public async Task<Dictionary<string, bool>> MarketsStatus([FromBody] string[] marketIds = null)
+        {
+            var allMarkets = await _marketRepository.GetAsync();
+            if (marketIds == null || !marketIds.Any())
+            {
+                marketIds = allMarkets.Select(x => x.Id).ToArray();
+            }
+            else
+            {
+                foreach (var marketId in marketIds)
+                {
+                    if (allMarkets.Select(x => x.Id).ToHashSet().Contains(marketId))
+                    {
+                        continue;
+                    }
+                        
+                    throw new ArgumentException($"Market {marketId} does not exist", nameof(marketIds));

```

Comment:
Not sure about the exception, we could just filter out not existing and return data only for existing ones. But not critical.

Reply:
Yep, I think it's not a big deal.